packages/nvidia_nat_core/src/nat/front_ends/fastapi/message_handler.py

```
@@ -175,10 +175,35 @@ async def __aenter__(self) -> "WebSocketMessageHandler":
+            return
+        async with self._session_manager.session(
+                http_connection=self._socket,
+                user_authentication_callback=self._flow_handler.authenticate,
+        ):
+            for name, cfg in self._session_manager.config.authentication.items():
+                if cfg.preflight_auth:
+                    logger.debug("Preflight auth: authenticating provider '%s'", name)
+                    provider = await self._session_manager.shared_builder.get_auth_provider(name)
+                    try:
+                        await provider.authenticate()
+                    except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)